src/app/InteractionModelEngine.cpp

```diff
@@ -1699,6 +1703,18 @@ CHIP_ERROR InteractionModelEngine::PushFront(SingleLinkedListNode<T> *& aObjectL
 void InteractionModelEngine::DispatchCommand(CommandHandlerImpl & apCommandObj, const ConcreteCommandPath & aCommandPath,
                                              TLV::TLVReader & apPayload)
 {
+#if CHIP_CONFIG_USE_DATA_MODEL_INTERFACE
+
+    DataModel::InvokeRequest request;
+    request.path = aCommandPath;
+
+    std::optional<DataModel::ActionReturnStatus> status = GetDataModelProvider()->Invoke(request, apPayload, &apCommandObj);
+
+    if (status.has_value())
```

Comment:
What is value() is CHIP_NO_ERROR (the equivalent that means OK for ActionReturnStatus) ?

Reply:
Updated documentation some more. std::nullopt is distinctly different from CHIP_NO_ERROR. Added notes both here and in the Provider.h documentation.